components/server/src/workspace/workspace-service.ts

```diff
@@ -61,11 +62,35 @@ export class WorkspaceService {
     async getWorkspace(userId: string, workspaceId: string): Promise<Workspace> {
         await this.auth.checkPermissionOnWorkspace(userId, "access", workspaceId);
 
-        const workspace = await this.db.findById(workspaceId);
-        if (!workspace || !!workspace.softDeleted || workspace.deleted) {
-            throw new ApplicationError(ErrorCodes.NOT_FOUND, "Workspace not found.");
+        return this.doGetWorkspace(workspaceId);
+    }
+
+    async getOwnerToken(userId: string, workspaceId: string): Promise<string> {
+        await this.auth.checkPermissionOnWorkspace(userId, "access", workspaceId);
+
+        // Check: is deleted?
+        await this.doGetWorkspace(workspaceId);
+
+        const latestInstance = await this.db.findCurrentInstance(workspaceId);
+        const ownerToken = latestInstance?.status.ownerToken;
+        if (!ownerToken) {
+            throw new ApplicationError(ErrorCodes.NOT_FOUND, "owner token not found");
         }
-        return workspace;
+        return ownerToken;
+    }
+
+    async getIDECredentials(userId: string, workspaceId: string): Promise<string> {
+        await this.auth.checkPermissionOnWorkspace(userId, "access", workspaceId);
+
+        return this.db.transaction(async (db) => {
```

Comment:
I there is no good reason to run things in a transaction, let's not do it.

Reply:
I think there is a good reason: Whenever we are updating deep fields, which requires us to "read + update", that should happen in a transaction, to avoid lost updates.
In this case one could argue that workspace.config is not update very often etc., but I would like to avoid that bath honestly.
I wonder what that field is used for in the first place, though. :see_no_evil: